### DISCUSSION TITLE
Smart card return SCARD_E_NO_READERS_AVAILABLE error

### DISCUSSION THREAD

**Discussion by hxf3751** [FreeRDP/FreeRDP#8005] [General]
Hi, I want to redirect smart card by freerdp from linux to win10. It works well from win10 to win10 by mstsc. But, It returns SCARD_E_NO_READERS_AVAILABLE error from linux to win10 by freerdp. PCSC can list the smart card device. My two commands are as follows:
xfreerdp /v:ip /u:admin /p:admin123 /smartcard:3A59,
or xfreerdp /v:ip /u:admin /p:admin123 /smartcard:,

The following images are some tip from pcsc and freerdp:

 pcsc started:
![pcsc_sscard start](https://user-images.githubusercontent.com/86779241/176372206-bb062b9a-ed67-4744-8b31-fab1044293fe.png)

freerdp connected:
![freerdp_sscard](https://user-images.githubusercontent.com/86779241/176372437-99cbef1a-b3f5-4ae8-a264-1bd28cbc8d35.png)

pcsc later:
![pcsc_sscard](https://user-images.githubusercontent.com/86779241/176372612-61654c39-3e03-4cba-92fa-3e1a56c36087.png)

**Comment by akallabeth** [comment]
your syntax is wrong. See `xfreerdp /help`: `/smartcard[:<str>[,<str>...]]     Redirect the smartcard devices containing`

* `xfreerdp /smartcard <other args>`  redirects all smartcards

**Reply by hxf3751** [reply]
I really don't know how to write the right syntax,  Could you help me to offer an example to redirect smart card for my device?   Thanks very much.  The device: VID: 0x3A59, PID: 0x445A, path: /dev/bus/usb/009/014.
/smartcard ?
/samrcard: /dev/bus/usb/009/014?
or others?

**Comment by hxf3751** [comment]
I use the command "xfreerdp /v:ip /u:admin /p:admin123 /smartcard" to redirect all smard cards, but the error is same as before. Do you know why?

**Reply by akallabeth** [reply]
is your freerdp build with proper smartcard support? Check `buildconfig`

**Comment by hxf3751** [comment]
Yes, in code, status = g_PCSC.pfnSCardListReaders(hContext, mszGroups, mszReaders, &pcsc_cchReaders);  this line returned status is error. It seems that the "libpcsclite.so" cannot find the reader, but in pcsc service , the reader is exist.

**Reply by akallabeth** [reply]
@hxf3751 then your `pcdsc` is the source of the problem, check the configuration (or disable it as a test) and check if the library access works then.

**Comment by hxf3751** [comment]
If I close the pcsc, freerdp will return error "SCARD_E_NO_SERVICE".

**Reply by akallabeth** [reply]
ok, then your `pcsc` configuration seems broken.
we can only return what the `API` returns, no way to configure it from our side.
